connector/connect/common/src/main/protobuf/spark/connect/relations.proto

```diff
@@ -140,6 +141,21 @@ message Read {
     // (Optional) A list of path for file-system backed data sources.
     repeated string paths = 4;
   }
+
+  message PartitionedJDBC {
+    // (Required) JDBC URL.
+    string url = 1;
+
+    // (Required) Name of the table in the external database.
+    string table = 2;
+
+    // (Optional) Condition in the where clause for each partition.
+    repeated string predicates = 3;
```

Comment:
Can we just put the predicates into the DataSource message?

Reply:
But the transform path is very different from DataSource.

Reply:
I think it's simple to add a `if-else` in `transformReadRel`, if we can reuse existing DataSource message (with new field  `predicates `)

Reply:
OK. Let's put the predicates into the DataSource message.